hermes-console-vue/src/views/admin/constraints/create-constraint-form/CreateConstraintFormView.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { describe, expect } from 'vitest';
+import { render, renderWithEmits } from '@/utils/test-utils';
+import { waitFor } from '@testing-library/vue';
+import CreateConstraintFormView from '@/views/admin/constraints/create-constraint-form/CreateConstraintFormView.vue';
+
+describe('CreateConstraintForm', () => {
+  it('should emit a cancel event when user clicks cancel button', async () => {
+    // given
+    const wrapper = renderWithEmits(CreateConstraintFormView, {
+      props: {
+        subscription: false,
+      },
+    });
+
+    await wrapper
+      .find('[data-testid="createConstraintCancel"]')
+      .trigger('click');
+
+    await waitFor(() => {
+      const cancelEvents = wrapper.emitted('cancel')!!;
+      expect(cancelEvents).toHaveLength(1);
+    });
+  });
+
+  it('should render subscription name field when in subscription mode', async () => {
+    // given
+    const { queryAllByText } = render(CreateConstraintFormView, {
+      props: {
+        subscription: true,
+      },
+    });
+    // when
```

Comment:
I think it should be when/then instead of given/when

Reply:
fixed